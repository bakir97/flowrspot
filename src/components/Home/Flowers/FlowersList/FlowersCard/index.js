import FlowersCard from "./FlowersCard";
export default FlowersCard;
