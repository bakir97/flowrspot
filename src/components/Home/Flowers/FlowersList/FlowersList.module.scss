.container {
  margin: auto;
  padding-left: 20px;
  margin-top: 90px;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  @media screen and (max-width: 768px) {
    padding: 0 16px;
    justify-content: space-around;
  }
  @media screen and (max-width: 600px) {
    padding: 0 16px;
    justify-content: space-around;
  }
}
