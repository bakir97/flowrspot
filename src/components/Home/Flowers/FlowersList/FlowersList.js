import React from "react";
import FlowersCard from "./FlowersCard";
import styles from "./FlowersList.module.scss";
const FlowersList = ({ flowers, user, addFavFlower }) => {
  const flowersRender = () =>
    flowers.map(flower => (
      <FlowersCard
        key={flower.id}
        flower={flower}
        user={user}
        addFavFlower={addFavFlower}
      />
    ));
  return <div className={styles.container}>{flowersRender()}</div>;
};

export default FlowersList;
