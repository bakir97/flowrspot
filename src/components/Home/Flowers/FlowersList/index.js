import FlowersList from "./FlowersList";
export default FlowersList;
