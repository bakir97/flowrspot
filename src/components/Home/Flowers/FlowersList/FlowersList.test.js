import React from "react";
import { shallow } from "enzyme";
import FlowersList from "./FlowersList";
describe("FlowersList", () => {
  it("should render correctly", () => {
    const component = shallow(<FlowersList flowers={[]} />);
    expect(component).toMatchSnapshot();
  });
  it("should render items", () => {
    const flowers = [{ id: "1" }, { id: "2" }, { id: "3" }];
    const component = shallow(<FlowersList flowers={flowers} />);
    expect(component).toMatchSnapshot();
  });
});
