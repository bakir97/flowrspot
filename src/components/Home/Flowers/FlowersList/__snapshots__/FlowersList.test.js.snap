// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowersList should render correctly 1`] = `
<div
  className="container"
/>
`;

exports[`FlowersList should render items 1`] = `
<div
  className="container"
>
  <FlowersCard
    flower={
      Object {
        "id": "1",
      }
    }
    key="1"
  />
  <FlowersCard
    flower={
      Object {
        "id": "2",
      }
    }
    key="2"
  />
  <FlowersCard
    flower={
      Object {
        "id": "3",
      }
    }
    key="3"
  />
</div>
`;
